pkg/oci/layout/image.go

```diff
@@ -0,0 +1,103 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package layout
+
+import (
+	"fmt"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/layout"
+	"github.com/google/go-containerregistry/pkg/v1/partial"
+	"github.com/sigstore/cosign/pkg/oci"
+	"github.com/sigstore/cosign/pkg/oci/siglayer"
+)
+
+// SignedImage provides access to a remote image reference, and its signatures.
+func SignedImage(path string) (oci.SignedImage, error) {
+	p, err := layout.FromPath(imagePath(path))
```

Comment:
Is this ~always an index?  If so, I'm wondering if we should only expose `SignedIndex` 🤔  (or at least start there instead, and add "sugar" later)

Reply:
The index is the entrypoint into this layout -- you could interpret the entire layout as a single index and push that to a registry, but I think it's more interesting to just use this index data structure as a list of what's contained within the directory. You can have a bunch of cosign-isms around annotations on the top-level `index.json` index that are meaningful to cosign without them leaking from this implementation.

Reply:
It might be interesting to have an idea of a "snapshot" of a `SignedImage`, that is implemented as an index that points to every piece of relevant metadata?